src/scala/ly/stealth/mesos/kafka/Rebalancer.scala

```diff
@@ -121,6 +144,36 @@ class Rebalancer {
     reassignment.toMap
   }
 
+  private def getRealignments(brokerIds: Seq[Int], topics: util.List[String], assignment: Map[TopicAndPartition, Seq[Int]], replicas: Int = -1, fixedStartIndex: Int = -1): Map[TopicAndPartition, Seq[Int]] = {
+    var realignment : Map[TopicAndPartition, Seq[Int]] = new mutable.HashMap[TopicAndPartition, List[Int]]()
+
+    val rand = new Random
+    var replacementReplicaIndex = if (fixedStartIndex != -1) fixedStartIndex else rand.nextInt(brokerIds.size)
+    assignment.foreach { entry =>
+      val topic = entry._1
+      val members = entry._2
+      val rf: Int = if (replicas != -1) replicas else members.size
+
+      if((rf != members.size) || ((brokerIds intersect members).size != members.size)) {
```

Comment:
Can you toss a couple comments in here (particularly describing the logic)?  It's really hard to figure out what its doing.